examples/light-switch-app/nrfconnect/main/AppTask.cpp

```diff
@@ -259,6 +266,11 @@ void AppTask::ButtonReleaseHandler(AppEvent * aEvent)
             sAppTask.CancelTimer(Timer::DimmerTrigger);
             sWasDimmerTriggered = false;
             break;
+        case GENERIC_SWITCH_BUTTON:
+            LOG_INF("GenericSwitch: ShortRelease");
+            sAppTask.StartTimer(Timer::Function, 0);
```

Comment:
Using a zero timeout function seems a bit overcomplicated. Consider adding another event type in `AppEvent.h`.

Reply:
Agree, that I do not need the `StartTimer()`, but I don't think we need another event type (kEventType_Button seems fine). Shall I directly call `LightSwitch::GetInstance().GenericSwitchInitialPress();`/`LightSwitch::GetInstance().GenericSwitchReleasePress();` in `ButtonReleaseHandler()`/`ButtonPushHandler()` or is that the wrong context?

Reply:
You're right, `kEventType_Button` is enough and you can call `GenericSwitchInitialPress/ReleasePress` directly. However, in those methods you can modify the attributes using:
```
DeviceLayer::SystemLayer().ScheduleLambda([] {
  uint8_t newPosition = 1;

  Clusters::Switch::Attributes::CurrentPosition::Set(mLightGenericSwitchEndpointId, newPosition);
  Clusters::SwitchServer::Instance().OnInitialPress(mLightGenericSwitchEndpointId, newPosition);
});
```
to make sure that CHIP data is modified on the CHIP thread.